backend/http/writer.go

```diff
@@ -0,0 +1,62 @@
+package http
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+)
+
+// chunkSize is the default chunk size for the flushResponseWriter.
+const chunkSize = 4096
+
+// flushResponseWriter is a wrapper around http.ResponseWriter that handles
+// buffered output. It chunks the output and flushes it to the client. It
+// implements the io.ReaderFrom interface to read from an io.Reader and write
+// to the ResponseWriter. It also implements a no-op Close method to satisfy
+// the io.Closer interface.
+// Useful when using proxies.
+type flushResponseWriter struct {
+	http.ResponseWriter
+	log *log.Logger
+}
+
+// ReadFrom implements io.ReaderFrom interface.
+func (f *flushResponseWriter) ReadFrom(r io.Reader) (int64, error) {
+	flusher := http.NewResponseController(f.ResponseWriter) // nolint: bodyclose
+
+	var n int64
+	p := make([]byte, chunkSize)
```

Comment:
Maybe make chunk size configurable?

Reply:
Make it at least public.

Reply:
There may be heuritistics known by the backend that could change this to an optimal value to optimise an specific use-case. 

Instead of making it public, let's make it a field of `flushResponseWriter`. If needed, we could then expose it via variadic option of the Handler.